src/UriNormalize.c

```diff
@@ -152,7 +150,13 @@ static URI_INLINE void URI_FUNC(PreventLeakage)(URI_TYPE(Uri) * uri,
 		}
 	}
 
-	/* NOTE: Port cannot happen! */
+	if (revertMask & URI_NORMALIZE_PORT) {
+		if (uri->portText.first != uri->portText.afterLast) {
+			memory->free(memory, (URI_CHAR *)uri->portText.first);
+		}
+		uri->portText.first = NULL;
+		uri->portText.afterLast = NULL;
+	}
```

Comment:
I would prefer to have this code go [into a new function](https://github.com/uriparser/uriparser/pull/230#discussion_r2158710056) that calls out to `PreventLeakage` and does additional things on top. This way we will have code in there that is dead to the eyes of normalization.